app/src/main/java/org/xbmc/kore/ui/BaseActivity.java

```diff
@@ -15,32 +15,88 @@
  */
 package org.xbmc.kore.ui;
 
+import android.content.Intent;
 import android.content.SharedPreferences;
 import android.os.Bundle;
 import android.preference.PreferenceManager;
 import android.support.v7.app.AppCompatActivity;
+import android.view.KeyEvent;
 
 import org.xbmc.kore.Settings;
+import org.xbmc.kore.host.HostManager;
+import org.xbmc.kore.jsonrpc.method.Application;
+import org.xbmc.kore.jsonrpc.type.GlobalType;
+import org.xbmc.kore.ui.sections.hosts.AddHostActivity;
 import org.xbmc.kore.utils.UIUtils;
 
 /**
  * Base activity, where common behaviour is implemented
  */
 public abstract class BaseActivity extends AppCompatActivity {
 
+	/**
+	 * Host manager singleton
+	 */
+	protected HostManager hostManager = null;
+
 	@Override
 	protected void onCreate(Bundle savedInstanceState) {
         SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
         setTheme(UIUtils.getThemeResourceId(
                 prefs.getString(Settings.KEY_PREF_THEME, Settings.DEFAULT_PREF_THEME)));
         super.onCreate(savedInstanceState);
+
+		hostManager = HostManager.getInstance(this);
+
+		// Check if we have any hosts setup
+		if (hostManager.getHostInfo() == null && !(this instanceof AddHostActivity)) {
+			final Intent intent = new Intent(this, AddHostActivity.class);
+			intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+			intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+			startActivity(intent);
+			finish();
+		}
```

Comment:
This seems to be useful, but how did you come up with it? Did you ran into a situation where this happened to you? If so can you describe it.

Reply:
Forget the comment. I see that you migrated this from RemoteActivity. Anyway, the question still stands: did you ran into a situation where that happened (and the remote activity didn't handle it)? Guess that removing all hosts and then going into another activity might do that, but need to check.

Reply:
Hi!

I'm not quite sure I follow you. What do you mean by: 

> This happened to you?

What is "this"?

The code change here are more of a refactor, all of this was existing code that I've just moved around in order to allow volume keys to control Kodi's volume from anywhere within Kore. The only thing I've added is:

` && !(this instanceof AddHostActivity)` since clearly we'd not want an infinite loop in case of the `AddHostActivity`

#453 requested this change and I'd personally would like to use volume buttons everywhere too.

Reply:
Ok. This particular code isn't related to the volume control functions, so i wondered why it was moved.

It starts the add host activity if there's no hosts configured, so it's a safeguard. I was asking if you moved it because a specific situation happened to you and moving it solved the issue.

Reply:
`hostManager` is used inside `dispatchKeyEvent`, this was a way to make it visible for that method.